src/siphash24.rs

```diff
@@ -135,14 +134,23 @@ impl HashEngine {
     }
 }
 
-impl Write for HashEngine {
-    #[inline]
-    fn flush(&mut self) -> ::std::io::Result<()> {
-        Ok(())
+impl EngineTrait for HashEngine {
+    type MidState = State;
+
+    #[cfg(not(feature = "fuzztarget"))]
+    fn midstate(&self) -> State {
```

Comment:
Although this seems to be just a move I'd be interested if the conditional compilation makes any sense here (I don't see the point of it).

Reply:
Heh, yeah, I'll get rid of it. It does look weird.